packages/roosterjs-editor-core/lib/editor/Editor.ts

```diff
@@ -345,16 +343,10 @@ export default class Editor {
         triggerExtractContentEvent: boolean = true,
         includeSelectionMarker: boolean = false
     ): string {
-        let contentDiv = this.core.contentDiv;
-        let content = contentDiv.innerHTML;
-        let selectionPath: SelectionPath;
-
-        if (
-            includeSelectionMarker &&
-            (selectionPath = getSelectionPath(contentDiv, this.getSelectionRange()))
-        ) {
-            content += `<!--${JSON.stringify(selectionPath)}-->`;
-        }
+        let content = getHtmlWithSelectionPath(

```

Comment:
curious why you decided to factor this out? I think it looked more clear when it was all written here